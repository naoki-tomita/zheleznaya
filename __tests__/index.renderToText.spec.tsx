declare const React = ""; // suppress error.
import { renderToText, h, Component } from "../index";

describe("renderToText", () => {
  it("should rendered string match to snapshots.", () => {
    const someString = "someString";
    const Inner: Component = (props: { name: string }, children) => {
      return (
        <div>
        child component {props.name}
        <div>{children}</div>
        </div>
      );
    }
    const App = () => {
      return (
        <div id="root">
          <h1>Hello world</h1>
          <h2 style={{ display: "flex", justifyContent: "center" }}>Style</h2>
          <h3 class="test-foo">className</h3>
          <h4>{someString}</h4>
          <div ref={el => el.innerHTML = "<span>ref innerHTML</span>"} />
          <Inner name="props.name" >Hello <div>world</div></Inner>
        </div>
      );
    }

    expect(renderToText(<App />)).toMatchSnapshot();
  })
});
