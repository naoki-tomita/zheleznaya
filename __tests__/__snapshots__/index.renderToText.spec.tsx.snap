// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderToText should rendered string match to snapshots. 1`] = `"<div id=\\"root\\"><h1 >Hello world</h1><h2 style=\\"display=flex;justifyContent=center;\\">Style</h2><h3 class=\\"test-foo\\">className</h3><h4 >someString</h4><div ref=\\"\\"><span>ref innerHTML</span></div><div >child component props.name<div >Hello <div >world</div></div></div></div>"`;
